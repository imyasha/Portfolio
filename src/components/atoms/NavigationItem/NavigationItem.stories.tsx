import { NavigationItem, NavigationItemProps } from './NavigationItem';

export default {
	title: 'atoms/NavigationItem',
	component: NavigationItem,
};

export const NavigationItemStory = (args: NavigationItemProps) => (
	<NavigationItem {...args} />
);

NavigationItemStory.storyName = 'NavigationItem';
NavigationItemStory.args = {
	href: '/',
	title: 'Navigation Link',
};
