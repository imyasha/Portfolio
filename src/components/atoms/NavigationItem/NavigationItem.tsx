import classNames from 'classnames';
import Link from 'next/link';
import { useRouter } from 'next/router';
import { motion, Variants } from 'framer-motion';

export interface NavigationItemProps {
	href: string;
	title: string;
	variants: Variants;
	initial: string;
	animate: string;
	customDelay: number;
}

const NavigationItem = ({
	href,
	title,
	variants,
	initial,
	animate,
	customDelay,
}: NavigationItemProps) => {
	const { asPath } = useRouter();
	const isActive = asPath.startsWith(href);

	return (
		<motion.li
			variants={variants}
			initial={initial}
			animate={animate}
			custom={customDelay}
		>
			<Link href={href}>
				<a
					className={classNames(
						isActive
							? 'font-bold text-off-black dark:text-off-white'
							: 'font-medium text-grey-700 md:text-grey-500 hover:text-off-black dark:hover:text-off-white',
						'md:underlined transition relative block whitespace-nowrap text-2xl md:text-lg'
					)}
				>
					{title}
				</a>
			</Link>
		</motion.li>
	);
};

export { NavigationItem };
