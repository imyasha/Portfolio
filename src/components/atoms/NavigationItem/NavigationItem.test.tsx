import { NavigationItem } from '../NavigationItem';
import { render } from '@testing-library/react';
import { useRouter } from 'next/router';

jest.mock('next/router', () => ({
	__esModule: true,
	useRouter: jest.fn().mockReturnValue({
		route: '',
		pathname: '',
		query: '',
		asPath: '',
	}),
}));

jest.mock('next/link', () => {
	const React = require('react');
	return ({ children, href }: React.PropsWithChildren<{ href: string }>) =>
		React.cloneElement(React.Children.only(children), { href });
});

describe('NavigationItem', () => {
	(useRouter as jest.Mock).mockReturnValue({
		asPath: '',
	});

	it('renders correctly', () => {
		const { container } = render(
			<NavigationItem
				href="/"
				title="Navigation Link"
				variants={{
					hidden: { opacity: 0, x: -50 },
					visible: (custom: number) => ({
						opacity: 1,
						x: 0,
						transition: { delay: custom },
					}),
				}}
				initial="initial"
				animate="animate"
				customDelay={0}
			/>
		);
		expect(container).toMatchSnapshot();
	});
});
