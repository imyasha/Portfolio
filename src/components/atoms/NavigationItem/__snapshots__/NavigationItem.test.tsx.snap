// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationItem renders correctly 1`] = `
<div>
  <li>
    <a
      class="font-medium text-grey-700 md:text-grey-500 hover:text-off-black dark:hover:text-off-white md:underlined transition relative block whitespace-nowrap text-2xl md:text-lg"
      href="/"
    >
      Navigation Link
    </a>
  </li>
</div>
`;
